components/builder-scheduler/src/error.rs

```diff
@@ -175,3 +179,9 @@ impl From<zmq::Error> for Error {
         Error::Zmq(err)
     }
 }
+
+impl From<depot_client::Error> for Error {
```

Comment:
While this pattern is littered through the codebase I would avoid it. This pattern creates a situation where we falsly report about an error because it acts as a catch all for anything `depot_client::Error`.

I would use `func().map_err(Error::PromoteFailure)?` instead of relying on this `From` trait. That will ensure that only depot_client errors which arise during a promote failure get turned into a promote failure.